datalad_deprecated/gitrepo.py

```diff
@@ -0,0 +1,247 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Internal low-level interface to Git repositories
```

Comment:
coverage reports that no lines in this file are covered although tests are there... because to make them in effect for here we need that PR https://github.com/datalad/datalad/pull/6010 to be merged/released first... before then we do not even know if this all works... I will sleep on it to see if there is some "simpleish" way (may be multisubclass  `DeprecatedGitRepoMethods` and `GitRepo` in the test_* here into a local to that module GitRepo? in one of the runs) so we could still test here and in datalad core...?

Reply:
I ran it locally, of course.

Reply:
And once this PR is merged, the PR in core will immediately be able to pick it up, unmerged.

Arguably, the cost of potentially adding broken code to an unreleased -deprecated extension is rather low. But again, can be verified locally.